interfaces/core.go

```diff
@@ -198,6 +198,10 @@ type StaticInfo struct {
 	// ImplicitOnClassic controls if a slot is automatically added to classic systems.
 	ImplicitOnClassic bool `json:"implicit-on-classic,omitempty"`
 
+	// DisruptiveForPlugOnRefresh tells if refreshing of a snap with a slot of this interface
+	// is disruptive for the snap on the plug side (when the interface is connected).
+	DisruptiveForPlugOnRefresh bool `json:"disruptive-for-plug,omitempty"`
```

Comment:
It's not very clear to me what "disruptive" means here, and how it should be handled. I see that this is handled by adding the connected snaps to the "affects" list (which again, I don't know what it does).

It would be great if this comment could be slightly expanded to make it more explicit what this disruption means in practice. I also wonder if `AffectsPlugOnRefresh` could be a better name (at least judging from how it's handled in the code).

Reply:
I agree that this probably should be expanded, mentioning affecting the mount namespace of the plug snap which requires freezing and thawing the running processes

Reply:
Thanks, I like `AffectsPlugOnRefresh`, renamed. Also enhanced/added the comments.

Reply:
Thanks!